src/jit/codegenxarch.cpp

```diff
@@ -2839,15 +2839,22 @@ CodeGen::genMultiRegCallStoreToLocal(GenTreePtr treeNode)
         {
             // targeReg[63:0] = targetReg[63:0]
             // targetReg[127:64] = reg1[127:64]
-            inst_RV_RV_IV(INS_shufpd, EA_16BYTE, targetReg, reg1, 0x10);
+            inst_RV_RV_IV(INS_shufpd, EA_16BYTE, targetReg, reg1, 0x00);
         }
         else 
         {
             assert(targetReg == reg1);
 
+            // We would need two shuffles to achieve this
+            // First:
+            // targeReg[63:0] = targetReg[63:0]
+            // targetReg[127:64] = reg0[63:0]
+            //
+            // Second:
             // targeReg[63:0] = targetReg[127:64]
             // targetReg[127:64] = reg0[63:0]
-            inst_RV_RV_IV(INS_shufpd, EA_16BYTE, targetReg, reg0, 0x01);
+            inst_RV_RV_IV(INS_shufpd, EA_16BYTE, targetReg, reg0, 0x00);
+            inst_RV_RV_IV(INS_shufpd, EA_16BYTE, targetReg, targetReg, 0x01);
         }
     }
```

Comment:
I had to open up the architecture manual to understand this. Perhaps a few more comments would be good. In particular, it is confusing that you refer to reg0 in the comment for the second instruction, but it doesn't actually use reg0 (this confused me for awhile). I think in the comments you should introduce new names for the original values (e.g. src1 and src2), and then refer to those in the comments, and annotate with the actual reg numbers as well.